### ISSUE TITLE
upload_to_testflight: <h2>Not Found</h2></div>

### ISSUE THREAD

**Issue by daminik00** [fastlane/fastlane#13903]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

##### Complete output when running fastlane, including the stack trace and command used

<details>

```
  +--------------------------------+--------------------------------------+
[20:09:29]: <!DOCTYPE html><html lang="en">
<head>
<title></title>
<link rel="stylesheet" href="/styles/main.css">
<style type='text/css'>
    /*include ../../dist/styles/main.css*/
</style>
</head>

<body><div id="wrapper" class="j42">
<h2>Not Found</h2></div>

<div class="footer">
    <p>Copyright &copy; <span id='year'>2015</span> Apple Inc. All rights reserved.</p>
    <ul><li><a href="https://www.apple.com/legal/internet-services/itunes/">Terms of Service</a></li><li><a href="https://www.apple.com/legal/privacy">Privacy Policy</a></li></ul>
</div>


<script type='text/javascript'>(function() {
    var isIos = navigator.userAgent.match(/(iPod|iPhone|iPad)/);
    document.body.setAttribute( 'class', (!isIos) ? 'otherBrowser' : 'isMobileSafari' )
    var yearEl = document.getElementById('year');
    if (yearEl) { yearEl.textContent = new Date().getFullYear(); }
})();
</script>

</body>
</html>

+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | default_platform       | 0           |
| 2    | latest_testflight_bui  | 13          |
|      | ld_number              |             |
| 3    | increment_build_numbe  | 1           |
|      | r                      |             |
| 4    | build_app              | 792         |
| 💥   | upload_to_testflight   | 3664        |
+------+------------------------+-------------+

[20:09:29]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

/Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/spaceship/lib/spaceship/test_flight/client.rb:330:in `handle_response': [!] <!DOCTYPE html><html lang="en"> (Spaceship::UnexpectedResponse)
<head>
<title></title>
<link rel="stylesheet" href="/styles/main.css">
<style type='text/css'>
    /*include ../../dist/styles/main.css*/
</style>
</head>

<body><div id="wrapper" class="j42">
<h2>Not Found</h2></div>

<div class="footer">
    <p>Copyright &copy; <span id='year'>2015</span> Apple Inc. All rights reserved.</p>
    <ul><li><a href="https://www.apple.com/legal/internet-services/itunes/">Terms of Service</a></li><li><a href="https://www.apple.com/legal/privacy">Privacy Policy</a></li></ul>
</div>


<script type='text/javascript'>(function() {
    var isIos = navigator.userAgent.match(/(iPod|iPhone|iPad)/);
    document.body.setAttribute( 'class', (!isIos) ? 'otherBrowser' : 'isMobileSafari' )
    var yearEl = document.getElementById('year');
    if (yearEl) { yearEl.textContent = new Date().getFullYear(); }
})();
</script>

</body>
</html>
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/spaceship/lib/spaceship/test_flight/client.rb:72:in `post_for_testflight_review'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/spaceship/lib/spaceship/test_flight/build.rb:217:in `submit_for_testflight_review!'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/pilot/lib/pilot/build_manager.rb:183:in `distribute_build'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/pilot/lib/pilot/build_manager.rb:108:in `distribute'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/pilot/lib/pilot/build_manager.rb:61:in `upload'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
	from Fastfile:38:in `block (2 levels) in parsing_binding'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /Users/daminik00/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.110.0/bin/fastlane:23:in `<top (required)>'
	from /Users/daminik00/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/daminik00/.fastlane/bin/bundle/bin/fastlane:22:in `<main>'
```
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.3                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.17.1 (Apple Git-112)          |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.14.3 (18D21c)                   |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 10.0                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  ENV["FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD"] = «"
  desc "Push a new beta build to TestFlight"

  lane :sendPush do
    changelog = File.read("Changelog.txt")
    url = URI.parse(URI.escape("http://api.daminik00.ru/push.php?info=#{changelog}"))
    req = Net::HTTP::Get.new(url.to_s)
    res = Net::HTTP.start(url.host, url.port) {|http|
      http.request(req)
    }
  end

  lane :beta do
    increment_build_number(
      build_number: latest_testflight_build_number.to_i + 1,
      xcodeproj: "SupremeBotiOSAcute.xcodeproj"
    )
    build_app(workspace: "SupremeBotiOSAcute.xcworkspace", scheme: "SupremeBotiOSAcute")
    changelog = File.read("Changelog.txt")
    upload_to_testflight(changelog: changelog, beta_app_description: "Удачных вам дропов", groups: "Me")
    sendPush
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("ru.daminik00.Supreme-Bot") # The bundle identifier of your app
apple_id("dev.daminik00@gmail.com") # Your Apple email address

itc_team_id("118705364") # App Store Connect Team ID
team_id("4X4HAH2D2G") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.110.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.7.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| public_suffix             | 2.0.5        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.0        |
| tty-spinner               | 0.9.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| commander-fastlane        | 4.4.6        |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| gh_inspector              | 1.1.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| mime-types                | 3.1          |
| signet                    | 0.11.0       |
| memoist                   | 0.16.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.2.7        |
| digest-crc                | 0.4.1        |
| google-cloud-storage      | 1.15.0       |
| emoji_regex               | 0.1.1        |
| highline                  | 1.7.10       |
| excon                     | 0.62.0       |
| CFPropertyList            | 3.0.0        |
| libxml-ruby               | 3.0.0        |
| plist                     | 3.4.0        |
| fastimage                 | 2.1.5        |
| addressable               | 2.5.2        |
| retriable                 | 3.1.2        |
| mime-types-data           | 3.2018.0812  |
| multi_json                | 1.13.1       |
| jwt                       | 2.1.0        |
| os                        | 1.0.0        |
| psych                     | 2.0.8        |
| uber                      | 0.1.0        |
</details>

*generated on:* **2018-12-13**
</details></pre>
</details>

**Comment by AlessandroFBK** [comment]
I got the same problem.

**Comment by davidbiga** [comment]
I am also experiencing the same issue.

**Comment by duyleekun** [comment]
I already had `ITSAppUsesNonExemptEncryption` key
```
<key>ITSAppUsesNonExemptEncryption</key>
<false/>
```

**Comment by tonyzp** [comment]
+1

**Comment by OUrsus** [comment]
The problem is reproduced for all our applications. `upload_to_testflight` is completely broken.

**Comment by janpio** [comment]
Quick summary of what has been posted: You are using `upload_to_testflight`as you have many times before, but now this crashes with an output of a HTML page that includes "Not Found" and a stacktrace similar to https://github.com/fastlane/fastlane/issues/13903#issuecomment-447097174. All correct?

---

I looked through the (obvious) stacktrace: It is successfully uploading the build, but when trying to `distribute` it does a `POST` request to a URL ending in `/review` (for that specific team, app, build) and gets this very much unexpected "Not Found".

---

Can someone please post the _complete_  output of fastlane (with `--verbose`) including the HTML and stacktrace? The parts _before_ it actually crashes are also relevant. There is some error catching logic in the code, and I need to know if and how this is triggered.

As you can see in the Stacktrace, the component crashing here when doing the actual request is spaceship. Spaceship has extensive logging to `/tmp/spaceship*.log`: https://github.com/fastlane/fastlane/blob/master/spaceship/docs/Debugging.md#logging Can someone please post the logs of such a failed upload attempt? This will tell us about the HTTP header being returned and if previous request may be failing or returning unexpected stuff as well.

Thanks 💯

**Comment by duyleekun** [comment]
> Do you see the upload in Apple's UI anywhere?
> It's possible Apple changed something on their end - we will of course try to adapt to this asap.

@janpio The build is on the iTunes Connect alright. It's just failure during Testflight distribution

**Comment by NicosKaralis** [comment]
@janpio is this the log you requested https://gist.github.com/NicosKaralis/efe0acf0a156caf5f2a80802dd666db3 ?

**Comment by janpio** [comment]
Yep, correct folder and type of log - but wrong file (or the logging is broken there as well). There should be at least one `POST` request to `providers/#{team_id}/apps/#{app_id}/builds/#{build_id}/review` (`#{...}` replaced with your IDs of course).

**Comment by NicosKaralis** [comment]
Updated the link

https://gist.github.com/NicosKaralis/efe0acf0a156caf5f2a80802dd666db3

**Comment by duyleekun** [comment]
Same kind of log
https://gist.github.com/duyleekun/1fc6a97422cd4b7c100b048194e49f2a

**Comment by janpio** [comment]
Yep, those are perfect. Will look into those. Thanks!

**Comment by duyleekun** [comment]
Can't find `providers/#{team_id}/apps/#{app_id}/builds/#{build_id}/review` in https://developer.apple.com/documentation/appstoreconnectapi/submit_an_app_for_beta_review

Seems like Apple deprecated the old endpoint?

**Comment by janpio** [comment]
Fastlane/spaceship is not using the App Store Connect API (yet).

**Comment by joshdholtz** [comment]
Hello everyone! I want to give an update. I was able to make some code changes and I got a submit working on my local machine again 😊 Bad news is we had to implement some of App Store Connect's new APIs so it will be a bit longer until I can properly put all the pieces together so that we can ship this fix but I was able to put the pieces back together.

Stay tuned 💪

**Comment by Enrico204** [comment]
Thanks guys :muscle:

**Comment by joshdholtz** [comment]
Heyyyy 👋 I think I got a fix working in PR #13919 if anybody would like to give this a shot 😊 

## How To
Replace `gem "fastlane"` in your `Gemfile with the following and run `bundle install`
```rb
gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "joshdholtz-post_for_testflight_review-new-api"
```

This uses the new App Store Connect API (the first use of it in _fastlane_) so having some other people testing this would be 💯

**Comment by duyleekun** [comment]
@joshdholtz On first trial, I'm getting following error. I will test again to confirm whether the error is from fastlane. FYI, I can find the build in the itunes website

```
[09:35:28]: ----------------------------------
[09:35:28]: --- Step: upload_to_testflight ---
[09:35:28]: ----------------------------------
[09:35:28]: Login to App Store Connect (developer@tripath.vn)
Available session is not valid any more. Continuing with normal login.
[09:35:31]: Login successful
[09:35:31]: Ready to upload new build to TestFlight (App: 1446885215)...
[09:35:32]: Going to upload updated app to App Store Connect
[09:35:32]: This might take a few minutes. Please don't interrupt the script.
[09:36:55]: iTunes Transporter successfully finished its job
[09:36:55]: --------------------------------------------------------------------
[09:36:55]: Successfully uploaded package to App Store Connect. It might take a few minutes until it's visible online.
[09:36:55]: --------------------------------------------------------------------
[09:36:55]: Successfully uploaded the new binary to App Store Connect
[09:36:55]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
+------------------------------------+-----------------------------------------------------------------------------------------------+
|                                                            Lane Context                                                            |
+------------------------------------+-----------------------------------------------------------------------------------------------+
| PLATFORM_NAME                      | ios                                                                                           |
| LANE_NAME                          | ios deploy_beta                                                                               |
| ORIGINAL_DEFAULT_KEYCHAIN          | "/Users/tripath/Library/Keychains/login.keychain-db"                                          |
| FL_CHANGELOG                       | Version Bump                                                                                  |
|                                    | Use version of fastlane that use itunes connect api                                           |
|                                    | Bump version string                                                                           |
|                                    | Version Bump                                                                                  |
|                                    | Add missing plist files                                                                       |
|                                    | D not update repo                                                                             |
|                                    | Test legacy                                                                                   |
|                                    | Version Bump                                                                                  |
|                                    | Add bundle install                                                                            |
|                                    | Version Bump                                                                                  |
| SIGH_PROFILE_TYPE                  | app-store                                                                                     |
| MATCH_PROVISIONING_PROFILE_MAPPING | {"vn.tripath.m090rn"=>"match AppStore vn.tripath.m090rn"}                                     |
| IPA_OUTPUT_PATH                    | /Users/tripath/builds/m090/RNApp/M090.ipa                                                     |
| DSYM_OUTPUT_PATH                   | /Users/tripath/builds/m090/RNApp/M090.app.dSYM.zip                                            |
| XCODEBUILD_ARCHIVE                 | /Users/tripath/Library/Developer/Xcode/Archives/2018-12-14/M090 2018-12-14 09.29.08.xcarchive |
+------------------------------------+-----------------------------------------------------------------------------------------------+
[09:36:57]: Could not find a build for app: 1446885215 on platform: ios

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | setup_circle_ci            | 0           |
| 2    | yarn                       | 55          |
| 3    | cocoapods                  | 122         |
| 4    | changelog_from_git_commits | 0           |
| 5    | match                      | 2           |
| 6    | build_app                  | 386         |
| 💥   | upload_to_testflight       | 89          |
+------+----------------------------+-------------+

[09:36:57]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
FastlaneCore::Interface::FastlaneCrash: [!] Could not find a build for app: 1446885215 on platform: ios
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane_core/lib/fastlane_core/ui/interface.rb:129:in `crash!'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane_core/lib/fastlane_core/build_watcher.rb:13:in `wait_for_build_processing_to_be_complete'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/pilot/lib/pilot/build_manager.rb:55:in `upload'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
  Fastfile:78:in `block (2 levels) in parsing_binding'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/lane.rb:33:in `call'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
  /usr/local/lib/ruby/gems/2.5.0/bundler/gems/fastlane-a469c588e701/bin/fastlane:23:in `<top (required)>'
  /usr/local/lib/ruby/gems/2.5.0/bin/fastlane:23:in `load'
  /usr/local/lib/ruby/gems/2.5.0/bin/fastlane:23:in `<top (required)>'
```

**Comment by duyleekun** [comment]
For some reason, 2nd time is a charm 👍 

```
[09:39:02]: ----------------------------------
[09:39:02]: --- Step: upload_to_testflight ---
[09:39:02]: ----------------------------------
[09:39:02]: Login to App Store Connect (developer@tripath.vn)
Available session is not valid any more. Continuing with normal login.
[09:39:05]: Login successful
[09:39:05]: Ready to upload new build to TestFlight (App: 1446885215)...
[09:39:06]: Going to upload updated app to App Store Connect
[09:39:06]: This might take a few minutes. Please don't interrupt the script.
[09:40:24]: iTunes Transporter successfully finished its job
[09:40:24]: --------------------------------------------------------------------
[09:40:24]: Successfully uploaded package to App Store Connect. It might take a few minutes until it's visible online.
[09:40:24]: --------------------------------------------------------------------
[09:40:24]: Successfully uploaded the new binary to App Store Connect
[09:40:24]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
[09:40:27]: Waiting for App Store Connect to finish processing the new build (2.0.0 - 35)
[09:40:58]: Waiting for App Store Connect to finish processing the new build (2.0.0 - 35)
[03:56:54]: Waiting for App Store Connect to finish processing the new build (2.0.0 - 35)
[03:57:24]: Waiting for App Store Connect to finish processing the new build (2.0.0 - 35)
[03:57:55]: Waiting for App Store Connect to finish processing the new build (2.0.0 - 35)
[03:58:25]: Waiting for App Store Connect to finish processing the new build (2.0.0 - 35)
[03:58:56]: Waiting for App Store Connect to finish processing the new build (2.0.0 - 35)
[03:59:27]: Build doesn't show up in the build list anymore, waiting for it to appear again (check your email for processing issues if this continues)
[03:59:57]: Build doesn't show up in the build list anymore, waiting for it to appear again (check your email for processing issues if this continues)
[04:00:28]: Build doesn't show up in the build list anymore, waiting for it to appear again (check your email for processing issues if this continues)
[04:00:59]: Build doesn't show up in the build list anymore, waiting for it to appear again (check your email for processing issues if this continues)
[04:01:29]: Build doesn't show up in the build list anymore, waiting for it to appear again (check your email for processing issues if this continues)
[04:02:00]: Successfully finished processing the build 2.0.0 - 35
[04:02:06]: Successfully set the changelog for build
[04:02:06]: Distributing new build to testers: 2.0.0 - 35
[04:02:07]: Successfully distributed build to External testers 🚀

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | setup_circle_ci            | 0           |
| 2    | yarn                       | 57          |
| 3    | cocoapods                  | 348         |
| 4    | changelog_from_git_commits | 0           |
| 5    | match                      | 2           |
| 6    | build_app                  | 370         |
| 7    | upload_to_testflight       | 66185       |
+------+----------------------------+-------------+

[04:02:07]: fastlane.tools just saved you 1116 minutes! 🎉
```

**Comment by joshdholtz** [comment]
@duyleekun Ohhhhh... I wonder if I also need to replace the API call that checks for build statuses as well 😩 I’m going to Disneyland with my nieces today but I will work on fixing that later tonight 💪

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by duyleekun** [comment]
@joshdholtz  The `Could not find a build for app` has happened to me before. It happened randomly so I guess Apple's at fault. Anyway, have a nice trip

**Comment by nickgsc** [comment]
@joshdholtz thanks for rolling such a quick fix for this. We tested out the branch and it allowed us to get past the waiting for review and get a build out into App Store Connect. Still got a failure related to an UnexpectedResponse, but it got far enough along to unblock us for now. This failure was on CircleCI:

```
bundler: failed to load command: fastlane (/Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bin/fastlane)
Spaceship::UnexpectedResponse: [!] The build is not in state for internal testing. - Build is not in internal testing state.
  /Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bundler/gems/fastlane-a469c588e701/spaceship/lib/spaceship/api/client.rb:84:in `handle_response'
  /Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bundler/gems/fastlane-a469c588e701/spaceship/lib/spaceship/api/client.rb:66:in `post_for_testflight_review'
  /Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bundler/gems/fastlane-a469c588e701/spaceship/lib/spaceship/test_flight/build.rb:223:in `submit_for_testflight_review!'
  /Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bundler/gems/fastlane-a469c588e701/pilot/lib/pilot/build_manager.rb:183:in `distribute_build'
  /Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bundler/gems/fastlane-a469c588e701/pilot/lib/pilot/build_manager.rb:108:in `distribute'
  /Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bundler/gems/fastlane-a469c588e701/pilot/lib/pilot/build_manager.rb:61:in `upload'
  /Users/distiller/gospotcheck-ios/vendor/bundle/ruby/2.5.0/bundler/gems/fastlane-a469c588e701/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
```

**Comment by nickgsc** [comment]
Update on my comment from last weekend. Today, one of our developers inadvertently reverted us back to the latest mainline fastlane version (2.111.0) and pushed a new build which failed. But what is weird is that we did **not** see the `<h2>Not Found</h2>` error from last week and the build did get uploaded successfully. Instead, we did get a slightly different error further down the build, trying to set the changelog, of all things:
```
[!] Could not set changelog: <html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 403 </title>
</head>
<body>
<h2>HTTP ERROR: 403</h2>
```

The fact that we are not getting the original error seems to suggest that Apple might have _un-deprecated_ the old iTunes connect endpoint? This changelog error appears to be something entirely new that we weren't seeing last week. I'll open a separate issue about that.

**Comment by udayasri** [comment]
I'm having the same `Not Found` issue in fastlane 2.111.0 . Did anyone able to get this done ? What is the state of PR #13919 ?

**Comment by janpio** [comment]
It has been merged but not released. You can test it with the instructions in https://github.com/fastlane/fastlane/issues/13903#issuecomment-447535838 to see if it fixes the issue for you.

**Comment by udayasri** [comment]
Thank you, will test it. Hopefully it will be released soon.

**Comment by davidbiga** [comment]
@janpio what is the timeline for getting this release out?  If there is anyway to get something pushed out today, that would be a major help!

**Comment by joshdholtz** [comment]
Release will be happening later today! Sorry to keep everyone waiting 😬 Testing out one more PR that fixes some adhoc profile stuff and then these will both be shipped out 🚀

**Comment by davidbiga** [comment]
@joshdholtz much appreciated.  Please let me know as soon as an update is released!

**Comment by janpio** [comment]
(2.112.0 is out 🚥 🚆 )

**Comment by davidbiga** [comment]
🙌 our build is processing right now.  best of luck 🤞

**Comment by VSAC-STORES-TEAM-INDIA** [comment]
We updated fastlane version to 2.111.0 and now we are not getting 'Not found' error but another error, same as what @nickgsc pointed out. Will this also be fixed in latest release?

/Users/ios-worker/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/bundler/gems/fastlane-c63166f9d98c/spaceship/lib/spaceship/connect_api/client.rb:84:in `handle_response': [!] The build is not in state for internal testing. - Build is not in internal testing state. (Spaceship::UnexpectedResponse) from /Users/ios-worker/.rbenv/versions/2.4.0/lib/ruby/gems/2.4.0/bundler/gems/fastlane-c63166f9d98c/spaceship/lib/spaceship/connect_api/client.rb:66:in `post_for_testflight_review' from /Users/ios-worker/.r

**Comment by udayasri** [comment]
Thanks a lot, it worked 👍

**Comment by janpio** [comment]
@VSAC-STORES-TEAM-INDIA Please open a new issue for this new problem - afaik we are not aware of this one yet, so nobody is working on understanding or fixing it. (Please make sure to include all the information the template asks for.)
